app/src/main/java/com/fstyle/structure_android/utils/navigator/Navigator.java

```diff
@@ -52,16 +67,107 @@ public void startActivityForResult(@NonNull Class<? extends Activity> clazz, Bun
         startActivityForResult(intent, requestCode);
     }
 
+    public void finishActivity() {
+        mActivity.finish();
+        setActivityTransactionAnimation(ActivityTransition.FINISH);
+    }
+
     public void finishActivityWithResult(Intent intent, int resultCode) {
         mActivity.setResult(resultCode, intent);
-        mActivity.finish();
+        finishActivity();
     }
 
     public void openUrl(String url) {
         if (TextUtils.isEmpty(url) || !Patterns.WEB_URL.matcher(url).matches()) {
             return;
         }
         Intent intent = new Intent(Intent.ACTION_VIEW).setData(Uri.parse(url));
-        mActivity.startActivity(intent);
+        startActivity(intent);
+    }
+
+    // Fragment
+
+    /**
+     * Go to next fragment which nested inside current fragment
+     *
+     * @param fragment new child fragment
+     */
+    public void goNextChildFragment(@IdRes int containerViewId, Fragment fragment,
+            boolean addToBackStack, int animation, String tag) {
+        FragmentTransaction transaction = mFragment.getChildFragmentManager().beginTransaction();
+        setFragmentTransactionAnimation(transaction, animation);
+        if (addToBackStack) transaction.addToBackStack(fragment.getClass().getSimpleName());
+        transaction.replace(containerViewId, fragment, tag);
+        transaction.commitAllowingStateLoss();
+        mFragment.getChildFragmentManager().executePendingTransactions();
+    }
+
+    /**
+     * Always keep 1 fragment in container
+     *
+     * @return true if fragment stack has pop
+     */
+    public boolean goBackChildFragment() {
+        boolean isShowPrevious = mFragment.getChildFragmentManager().getBackStackEntryCount() > 1;
+        if (isShowPrevious) {
+            mFragment.getChildFragmentManager().popBackStackImmediate();
+        }
+        return isShowPrevious;
+    }
+
+    private void setFragmentTransactionAnimation(FragmentTransaction transaction,
+            @NavigateAnim int animation) {
+        switch (animation) {
```

Comment:
WARNING: [SwitchIntDef] Switch statement on an `int` with known associated constant missing case `NavigateAnim.LEFT_RIGHT`
 This check warns if a `switch` statement does not explicitly include all the values declared by the typedef `@IntDef` declaration.